datasets/nell/nell.py

```diff
@@ -0,0 +1,218 @@
+"""
+
+
+_DESCRIPTION = """This dataset provides version 1115 of the belief
+extracted by CMU's Never Ending Language Learner (NELL) and version
+1110 of the candidate belief extracted by NELL. See
+http://rtw.ml.cmu.edu/rtw/overview.  NELL is an open information
+extraction system that attempts to read the Clueweb09 of 500 million
+web pages (http://boston.lti.cs.cmu.edu/Data/clueweb09/) and general
+web searches.
+
+The dataset has 4 configurations: nell_belief, nell_candidate,
+nell_belief_sentences, and nell_candidate_sentences. nell_belief is
+certainties of belief are lower. The two sentences config extracts the
+CPL sentence patterns filled with the applicable 'best' literal string
+for the entities filled into the sentence patterns. And also provides
+sentences found using web searches containing the entities and
+relationships.
+
+There are roughly 21M entries for nell_belief_sentences, and 100M
+sentences for nell_candidate_sentences.
+"""
+
+
+_LICENSE = """There does not appear to be a license for this dataset. It is made available by Carnegie Mellon University at http://rtw.ml.cmu.edu/rtw/resources
+"""
+
+_URLs = {
+    "nell_belief": "http://rtw.ml.cmu.edu/resources/results/08m/NELL.08m.1115.esv.csv.gz",
+    "nell_candidate": "http://rtw.ml.cmu.edu/resources/results/08m/NELL.08m.1110.cesv.csv.gz",
+    "nell_belief_sentences": "http://rtw.ml.cmu.edu/resources/results/08m/NELL.08m.1115.esv.csv.gz",
+    "nell_candidate_sentences": "http://rtw.ml.cmu.edu/resources/results/08m/NELL.08m.1110.cesv.csv.gz",
+}
+
+
+class Nell(datasets.GeneratorBasedBuilder):
+    """NELL dataset for knowledge bases and knowledge graphs and underlying sentences."""
+
+    VERSION = datasets.Version("0.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="nell_belief", description="The beliefs in raw data form", version="1115.0.0"),
+        datasets.BuilderConfig(
+            name="nell_candidate", description="The candidate beliefs in raw data form", version="1110.0.0"
+        ),
+        datasets.BuilderConfig(
+            name="nell_belief_sentences",
+            description="The underlying sentences available for the nell beliefs",
+            version="1115.0.0",
+        ),
+        datasets.BuilderConfig(
+            name="nell_candidate_sentences",
+            description="The underlying sentences available for the nell candidate beliefs",
+            version="1110.0.0",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "nell"
+
+    def _info(self):
+        if self.config.name in ("nell_belief", "nell_candidate"):
+            features = datasets.Features(
+                {
+                    "entity": datasets.Value("string"),
+                    "relation": datasets.Value("string"),
+                    "value": datasets.Value("string"),
+                    "iteration_of_promotion": datasets.Value("string"),
+                    "probability": datasets.Value("string"),
```

Comment:
this should be a float right ?

Reply:
Float will cause and error in PyArrow - exceeding 2GB, so I have to provide as a string: OverflowError: There was an overflow in the <class 'pyarrow.lib.FloatArray'>. Try to reduce writer_batch_size to have batches smaller than 2GB

See discussion with @yjernite  https://app.slack.com/client/T1RCG4490/activity-page/thread/C01FSCW6JLU-1607624760.211200. Don't know how to mark this as an issue for you to look at in future.

Reply:
Ok let's keep it this way for now then

You can open an issue that refers to this PR and with the error.
I can take a look at it later